src/main/java/com/cryptocurrency/newbchain/Block.java

```diff
@@ -0,0 +1,48 @@
+package com.cryptocurrency.newbchain;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import java.util.Date;
+
+/**
+* This class handles everything related to Blocks that will later constitute a Blockchain.
+*/
+@ToString
+@Getter
+@Setter
+public class Block {
+    
+    private String hash;
+	private String previousHash;
+	private String data; // data will be contain a summary of a transaction.
+	private long timeStamp; 
+    private int nonce;
+
+	public Block(String data, String previousHash) {
+		this.data = data;
+		this.previousHash = previousHash;
+		this.timeStamp = new Date().getTime();
+        this.hash = calculateBlockHash();
+	}
+
+    public String calculateBlockHash() {
```

Comment:
Maybe this method can be private

Reply:
done & done ;)

Reply:
Actually, the method is used in Main class to check the validity of the Blockchain. However, I reduced its visibility to package-private ;)

Reply:
ok, fine for me